### DISCUSSION TITLE
How to reset widget views

### DISCUSSION THREAD

**Discussion by Abhay0294** [eclipse-theia/theia#11966] [Q&A] [answered]
Hi,
I have the following requirement:

1.From the attached snap there are 3 widgets namely  widget 1, 2 & 3 , which are collapsible and expandable.
2. We're having a reset option under help in the top menu bar.
3. The requirement is when a user click on the reset, the widgets whichever are collapsed should get expanded. This is to render the standard view on the click of reset.
4. Also if a user has opened file explorer and if he/she clicks reset that should get collapsed and instead widget 1 should get expanded as standard view.

Requesting for the suggestions to implement the same.

<img width="957" alt="Requirement" src="https://user-images.githubusercontent.com/57217189/206101294-b1bc4738-4541-48dd-9146-4135dafa648e.png">

**Comment by msujew** [comment, marked as answer]
Hey @Abhay0294,

since Theia offers a very flexible workbench (i.e. all views can be drag&dropped, closed, etc) it might not be easily possible to programmatically reset the layout in the way you intend (i.e. instantaneously, with only certain widgets affected). One obvious approach would be to link your reset menu entry to the [`RESET_LAYOUT` command](https://github.com/eclipse-theia/theia/blob/85ca7ee1686aa86cef18abb94431f146a5b53641/packages/core/src/browser/shell/shell-layout-restorer.ts#L114-L118). Though that does automatically reload the application and basically bring the layout back into the initial state that every user sees the first time when they load your app.

**Reply by Abhay0294** [reply]
Thanks @msujew for the response. Will try to implement the same.

**Reply by Abhay0294** [reply]
@msujew just one thing I want to ask that , I think that reset feature is already implemented by theia, because whenever I close any one or all the widgets say widget 1, 2 &3 and then click on reset all views come to normal again in expanded view.

But my requirement is to expand all the collapsed widgets on the click of reset.

Please correct me if I'm wrong.

**Reply by vince-fugnitto** [reply]
@Abhay0294 I see two implementation which are probably interesting to you.

_**1** - default layout_:

Define the default layout for your application to be the 3 widgets you want by implementing `initializeLayout` (you can search for examples). It is possible you would need to extend other contributions to not implement the `initializeLayout` if you don't want them part of the default layout. Your reset can ultimately call `RESET_LAYOUT` command as Mark highlighted.

_**2** - programmatically hide/close and reveal_

The alternative would be to iterate over views and widgets in order to close or hide them and reveal the widgets you want at the locations you require. This can be done with `ApplicationShell` to get widgets.
